core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreWebViewClient.java

```diff
@@ -97,11 +101,21 @@ private String convertLegacyUrl(String url) {
   private boolean handleEpubAndPdf(String url) {
     String extension = MimeTypeMap.getFileExtensionFromUrl(url);
     if (DOCUMENT_TYPES.containsKey(extension)) {
-      Intent intent = new Intent(Intent.ACTION_VIEW);
-      Uri uri = Uri.parse(url);
-      intent.setDataAndType(uri, DOCUMENT_TYPES.get(extension));
-      intent.setFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
-      callback.openExternalUrl(intent);
+      File savedFile = FileUtils.downloadFileFromUrl(url, null, zimReaderContainer);
+      if (savedFile != null && savedFile.exists()) {
+        Uri uri = Build.VERSION.SDK_INT >= Build.VERSION_CODES.N
+          ? FileProvider.getUriForFile(
+          CoreApp.getInstance(),
```

Comment:
[nit] We use `CoreApp.getInstance()` multiple times. We should declare a variable here.

Reply:
ok